sympy/polys/polytools.py

```diff
@@ -5620,6 +5620,14 @@ def _symbolic_factor_list(expr, opt, method):
             if base.is_Number:
                 factors.append((base, exp))
                 continue
+        elif hasattr(arg, '_eval_factor'):
```

Comment:
Dropping the `else' leads to a collision for an earlier part of the `if`.  But the general idea is helpful.